src/serialization.cpp

```diff
@@ -62,36 +62,61 @@ Status serializeBlobToTensorProto(
     }
     }
     responseOutput.mutable_tensor_shape()->Clear();
-    for (auto dim : networkOutput->getEffectiveShape()) {
+    auto& effectiveNetworkOutputShape = networkOutput->getEffectiveShape();
+    auto& actualBlobShape = blob->getTensorDesc().getDims();
+    for (size_t i = 0; i < effectiveNetworkOutputShape.size(); ++i) {
+        size_t dim;
+        if (effectiveNetworkOutputShape[i] != 0) {
+            dim = effectiveNetworkOutputShape[i];
```

Comment:
We should validate blob precision and shape against tensorinfo precision and shape